### PR TITLE
Avoid throwing exception in LangageService for OE connections

### PR DESCRIPTION
**Pull request by kevcunnane**
The LanguageService was throwing multiple exceptions when trying to do a File.Open on the ObjectExplorer and Connection URIs. The fix was to add correct handling and only support relative paths, file:, untitled:, and tsqloutput: URI types since these represent actual files. Anything else will result in the parsing & binding being skipped.

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/Workspace/Workspace.cs [hunk 1]
```diff
@@ -259,6 +298,25 @@ internal string ResolveRelativeScriptPath(string baseFilePath, string relativePa
...
 
+        private bool IsNonFileUri(string path)
+        {
+            try
+            {
+
+                string scheme = GetScheme(path);
+                if (scheme != null && scheme.Length > 0 && !scheme.Equals("file"))
+                {
+                    return !fileUriSchemes.Contains(scheme);;
```
Comment (llali) [inline src/Microsoft.SqlTools.ServiceLayer/Workspace/Workspace.cs]: extra ;

Comment (llali) [inline src/Microsoft.SqlTools.ServiceLayer/Workspace/Workspace.cs]: >!fileUriSchemes.Contains(scheme) [](start = 27, length = 32)

!fileUriSchemes.Contains(scheme) already covers   !scheme.Equals("file")

File: src/Microsoft.SqlTools.ServiceLayer/Workspace/Workspace.cs [hunk 2]
```diff
@@ -142,17 +153,41 @@ private string ResolveFilePath(string filePath)
...
+            }
+            catch(Exception)
+            {
+                // Intentionally skipping if there is a parse error
+            }
+
+            return false;
+        }
+
+        public static string GetScheme(string uri)
```
Comment (llali) [inline src/Microsoft.SqlTools.ServiceLayer/Workspace/Workspace.cs]: just a suggestion ... these new methods are all for parsing a file path and can be all moved to a new class so the tests can only focus those methods.

File: src/Microsoft.SqlTools.ServiceLayer/Workspace/Workspace.cs [hunk 3]
```diff
@@ -259,6 +298,25 @@ internal string ResolveRelativeScriptPath(string baseFilePath, string relativePa
             return combinedPath;
         }
 
+        private bool IsNonFileUri(string path)
+        {
+            try
+            {
+
+                string scheme = GetScheme(path);
+                if (scheme != null && scheme.Length > 0 && !scheme.Equals("file"))
```
Comment (kburtram) [inline src/Microsoft.SqlTools.ServiceLayer/Workspace/Workspace.cs]: `scheme != null && scheme.Length > 0` could be shortened to `!string.IsNullOrEmpty(scheme)`

**Comment by kburtram** [comment]
@kevcunnane there's a test fail in AppVeyor.  Please check it's not related to this change.

**Comment by kevcunnane** [comment]
@benrr101 fully agree. Karl and I both discussed this and we need to refactor long term. We should be able to create specific connections first, and optionally kick off / allow for LanguageService / Metadata Services to be spun up for these. It's a refactor we'll need to take on at some point.

File: src/Microsoft.SqlTools.ServiceLayer/Workspace/Workspace.cs (hunk 2 above)
Comment (kevcunnane) [inline src/Microsoft.SqlTools.ServiceLayer/Workspace/Workspace.cs]: For now given time constraints will leave as-is

File: src/Microsoft.SqlTools.ServiceLayer/Workspace/Workspace.cs (hunk 1 above)
Comment (kevcunnane) [inline src/Microsoft.SqlTools.ServiceLayer/Workspace/Workspace.cs]: Thanks, will make these changes

**Pull request merged**